### ISSUE TITLE
Replace `mock_obj.assert_has_calls(calls)` with `assert mock_obj.mock_calls == calls`

### ISSUE THREAD

**Issue by harupy** [mlflow/mlflow#12480]
### Summary

[`assert_has_calls`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_has_calls) is not exact match and can lead to false positives:

```diff
diff --git a/tests/store/artifact/test_databricks_artifact_repo.py b/tests/store/artifact/test_databricks_artifact_repo.py
index 733db7bd3..5fc20d681 100644
--- a/tests/store/artifact/test_databricks_artifact_repo.py
+++ b/tests/store/artifact/test_databricks_artifact_repo.py
@@ -722,7 +722,7 @@ def test_list_artifacts_handles_pagination(databricks_artifact_repo):
             mock.call(ListArtifacts(run_id=MOCK_RUN_ID, path="", page_token="4")),
             mock.call(ListArtifacts(run_id=MOCK_RUN_ID, path="", page_token="6")),
         ]
-        message_mock.assert_has_calls(calls)
+        assert message_mock.mock_calls == calls
 
 
 def test_get_read_credential_infos_handles_pagination(databricks_artifact_repo):
```

In most cases, `assert mock_obj.mock_calls == calls` should work.

### Notes

- Make sure to open a PR from a **non-master** branch.
- Sign off the commit using the `-s` flag when making a commit:

  ```sh
  git commit -s -m "..."
           # ^^ make sure to use this
  ```

- Include `#{issue_number}` (e.g. `#123`) in the PR description when opening a PR.

**Comment by Jaishree2310** [comment]
Hey @harupy, can you please assign this issue to me?
Thanks!

**Comment by harupy** [comment]
@Jaishree2310 thanks!

**Comment by harupy** [comment]
Resolved by #12512
